server/service/src/processors/transfer/shipment/delete_inbound_shipment.rs

```diff
@@ -0,0 +1,77 @@
+use repository::{
+    InvoiceLineRowRepository, InvoiceRowRepository, InvoiceRowStatus, InvoiceRowType,
+    RepositoryError, StorageConnection,
+};
+
+use crate::invoice::common::get_lines_for_invoice;
+
+use super::{Operation, ShipmentTransferProcessor, ShipmentTransferProcessorRecord};
+
+const DESCRIPTION: &'static str =
+    "Delete inbound shipment when source outbound shipment is deleted";
+
+pub(crate) struct DeleteInboundShipmentProcessor;
+
+impl ShipmentTransferProcessor for DeleteInboundShipmentProcessor {
+    fn get_description(&self) -> String {
+        DESCRIPTION.to_string()
+    }
+
+    /// Inbound shipment will be deleted when all below conditions are met:
```

Comment:
Just to clarify the logic discussed in person, could consider adding an additional comment here something like  "Inbound shipments are automatically deleted when the associated outbound shipments are deleted."

Reply:
🙏 for reminder 4ae792e0be03a73cf3048c0a3451cc58ef0a52d6